web/api/v0/__init__.py

```diff
@@ -354,24 +354,31 @@ class MPSKSClientAddResource(Resource):
     )
     def post(self, user_id: int, password: str, mac: str, name: str) -> ResponseReturnValue:
         user = get_authenticated_user(user_id, password)
+        # checks rather the user has all settable mpsks clients created
+        if len(user.mpsk_clients) >= current_app.config.get("MAX_MPSKS", 30):
+            abort(400, message="User has the maximum count of mpsk clients.")
 
-        try:
-            # checks rather the user has all settable mpsks clients created
-            if len(user.mpsk_clients) > current_app.config.get("MAX_MPSKS", 30):
-                abort(400, message="User has the maximum count of mpsk clients.")
-
-            if not user.wifi_password:
-                abort(400, message="Please generate a wifi password first")
+        if not user.wifi_password:
+            abort(412, message="Please generate a wifi password first")
 
-            mpsk_client_create(session.session, owner=user, mac=mac, name=name, processor=user)
+        try:
+            mpsk_client = mpsk_client_create(
+                session.session, owner=user, mac=mac, name=name, processor=user
+            )
```

Comment:
Nice! I almost did this myself, but I'm glad you stumbled upon this and noted that the `if`s don't have to be in the `try` block.